azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/implementation/AppServiceManager.java

```diff
@@ -202,4 +201,105 @@ public FunctionApps functionApps() {
         }
         return functionApps;
     }
+
+    /**
+     *
+     * @return Returns the latest Windows runtime stacks
+     */
+    public Runtimes latestWindowsRuntimes() {
+        Hashtable<String, RuntimeVersion> variousRuntimes = new Hashtable<>();
+        variousRuntimes.put(JavaVersion.ComponentName.toLowerCase(), JavaVersion.OFF);
+        variousRuntimes.put(NodeVersion.ComponentName.toLowerCase(), NodeVersion.OFF);
+        variousRuntimes.put(WebContainer.ComponentName.toLowerCase(), WebContainer.OFF);
+        variousRuntimes.put(PythonVersion.ComponentName.toLowerCase(), PythonVersion.OFF);
+        variousRuntimes.put(PhpVersion.ComponentName.toLowerCase(), PhpVersion.OFF);
+        variousRuntimes.put(NetFrameworkVersion.ComponentName.toLowerCase(), NetFrameworkVersion.OFF);
+
+        Iterator<ApplicationStackInner> stackIter = this.innerManagementClient.providers().getAvailableStacks("Windows").iterator();
+        while (stackIter.hasNext()) {
+            ApplicationStackInner stack = stackIter.next();
+
+            RuntimeVersion runtime = variousRuntimes.get(stack.name().toLowerCase());
+            if (runtime != null) {
+                runtime.parseApplicationStackInner(stack);
+            } else {
+                //throw new FileNotFoundException("Runtime " + stack.name() + " not supported");
+            }
+        }
+
+        Runtimes runtimes = new Runtimes();
+
+        runtimes.withJavaVersions(JavaVersion.values());
+        runtimes.withNetFrameworkVersions(NetFrameworkVersion.values());
+        runtimes.withPhpVersions(PhpVersion.values());
+        runtimes.withPythonVersions(PythonVersion.values());
+        runtimes.withWebContainers(WebContainer.values());
+        runtimes.withNodeVersions(NodeVersion.values());
+
+        return runtimes;
+    }
+
+    public class Runtimes {
```

Comment:
Also good to have a more specific name instead of the generic name `Runtime` like AppServiceRuntimes. This way less confusion if there any other service that also has the concept of runtime.

Reply:
done